### ISSUE TITLE
default_branch is ignored and causes new project to break terraform execution

### ISSUE THREAD

**Issue by cmd-werner** [gitlabhq/terraform-provider-gitlab#630]
### Terraform Version

```shell
    terraform -version
    Terraform v0.15.4
    on darwin_amd64
    + provider registry.terraform.io/gitlabhq/gitlab v3.6.0
    + provider registry.terraform.io/hashicorp/aws v3.42.0
    + provider registry.terraform.io/hashicorp/external v2.1.0
    + provider registry.terraform.io/hashicorp/null v3.1.0
```

### Affected Resource(s)
- `gitlab_project`

### Terraform Configuration Files
```hcl
    resource "gitlab_project" "main" {
        name                                             = "approvals-issue"
        path                                             = "approvals-issue"
        visibility_level                                 = "private"
        namespace_id                                     = "namespace"
        archived                                         = false
        merge_requests_enabled                           = true
        approvals_before_merge                           = 2
        initialize_with_readme                           = true
        default_branch                                   = "master"
    }
```

### Debug Output
Shorten output:

```shell
$ terraform plan
resource "gitlab_project" "project" {
      + approvals_before_merge                           = 2
      + archived                                         = false
      + container_registry_enabled                       = false
      + http_url_to_repo                                 = (known after apply)
      + id                                               = (known after apply)
      + initialize_with_readme                           = true
      + issues_enabled                                   = true
      + lfs_enabled                                      = false
      + merge_method                                     = "merge"
      + merge_requests_enabled                           = true
      + mirror                                           = false
      + mirror_overwrites_diverged_branches              = false
      + mirror_trigger_builds                            = false
      + name                                             = "project"
      + namespace_id                                     = 1111111
      + only_allow_merge_if_all_discussions_are_resolved = true
      + only_allow_merge_if_pipeline_succeeds            = false
      + only_mirror_protected_branches                   = false
      + packages_enabled                                 = false
      + pages_access_level                               = "private"
      + path                                             = "project"
      + path_with_namespace                              = (known after apply)
      + pipelines_enabled                                = true
      + remove_source_branch_after_merge                 = true
      + request_access_enabled                           = true
      + runners_token                                    = (sensitive value)
      + shared_runners_enabled                           = false
      + snippets_enabled                                 = false
      + ssh_url_to_repo                                  = (known after apply)
      + use_custom_template                              = false
      + visibility_level                                 = "private"
      + web_url                                          = (known after apply)
      + wiki_enabled                                     = false
      + push_rules {
          + branch_name_regex = "(^(develop|dev|master|revert-[a-z0-9-]+)|(feature\\/|feat\\/|bug\\/|hotfix\\/|release\\/|rel\\/)[a-zA-Z0-9-._]+)$"
          + deny_delete_tag   = true
          + prevent_secrets   = true
        }
    }
```

```
$ terraform apply
....
╷
│ Error: PUT https://gitlab.com/api/v4/projects/27063723: 400 {message: {base: [Could not change HEAD: branch 'master' does not exist]}}
│ 
│   on resource "gitlab_project" "project" {
```

### Expected Behavior
Project is initialized with a README and its default branch is `master`

### Actual Behavior
Project is initialized with a README using `main` as default branch.

### Steps to Reproduce
1. Create a new `gitlab_project` with `default_branch` set to `master` and `initialize_with_readme` set to `true`
2. Run `terraform apply`

### References
- https://about.gitlab.com/blog/2021/03/10/new-git-default-branch-name/
- https://gitlab.com/gitlab-org/gitlab/-/issues/26261
- https://github.com/gitlabhq/terraform-provider-gitlab/issues/349

**Comment by abitrolly** [comment]
Yes, I guess you need to run `apply` twice for now.

Support for setting default branch for repo without commits had just landed in GitLab API - https://gitlab.com/gitlab-org/gitlab/-/issues/26261#note_601841507. It may work out of the box if you can repeat the experiment.

**Comment by callum-stakater** [comment]
running apply twice doesnt seem to work on latest gitlab provider, still seeing the same issue

**Comment by willianpaixao** [comment]
I can confirm this, having the same issue.
Default branch set and an _apply_ with no changes now fails.
My current workaround is to set the default branch on the group level.

**Comment by armsnyder** [comment]
This has been fixed and released in the latest provider 3.7.0 via #583
